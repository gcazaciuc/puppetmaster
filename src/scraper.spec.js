const Scraper = require('./scraper');
const CompoundComponentDetector = require('../src/compound-component-detector');
const ProjectFilesGenerator = require('../src/project-files-generator');
const ReactComponentGenerator = require('../src/react-component-generator');
const CodeGenerator = require('../src/code-generator');
const WordPOS = require('wordpos');

describe('Scraper component test suite', () => {
    let scraper = null;
    let projectFilesGenerator = null;
    let wordpos = null;
    let compoundDetector = null;
    let componentGenerator = null;
    let codeGenerator = null;
    beforeEach(() => {
        wordpos = new WordPOS();
        compoundDetector = new CompoundComponentDetector(wordpos);
        projectFilesGenerator = {
            generate: jest.fn(),
            generateComponent: jest.fn()
        };
        codeGenerator = new CodeGenerator();
        componentGenerator = new ReactComponentGenerator(
            projectFilesGenerator, 
            compoundDetector, 
            codeGenerator
        );
        scraper = new Scraper(projectFilesGenerator, componentGenerator);
    });

    it('Should properly parse the HTML and generate components', async () => {
        await scraper.scrape('file:///home/gabrielcazaciuc/react-puppet/fixtures/simple.html');
        const [firstCall] = projectFilesGenerator.generateComponent.mock.calls;
        expect(firstCall[0]).toBe('Application.js');
        expect(firstCall[1]).toMatchSnapshot();
    });
});