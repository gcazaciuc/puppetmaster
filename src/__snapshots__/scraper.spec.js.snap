// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scraper component test suite Should properly parse the HTML and generate components 1`] = `
"import React from 'react';
import styled from 'styled-components';

export const Application = props => {
    return (
        <div>
            <div className={'test'} />
        </div>
    );
};
"
`;

exports[`Scraper component test suite Should properly parse the HTML containing style tags and generate proper styled components 1`] = `
"import React from 'react';
import styled from 'styled-components';
const Div = styled.div\`
    color: rgb(0, 128, 0);
    margin-bottom: 20px;
    margin-left: 30px;
    margin-right: 30px;
    margin-top: 20px;
\`;
const Div1 = styled.div\`
    color: rgb(0, 0, 255);
    margin-bottom: 20px;
    margin-left: 30px;
    margin-right: 30px;
    margin-top: 20px;
\`;

export const Application = props => {
    return (
        <div>
            <Div1 className={'test'}>
                <Div />
            </Div1>
        </div>
    );
};
"
`;
